app/templating/metadata_context.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from app.libs.utils import convert_tx_id
+from app.utilities.date_utils import to_date
+
+logger = logging.getLogger(__name__)
+
+
+def build_metadata(schema_json, metadata):
```

Comment:
I don't think we should be passing schema_json to this method, it looks like it would be worth introducing an introduction context rather than overload the metadata context with it.